File: Segundo-Bimestre/Parcial 2/Unidad9-Exceptions/Unidad9-Exceptions_java/src/unidad9/exceptions/Ejercicio5.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package unidad9.exceptions;
import java.util.Scanner;
/**
 *
 * @author DELL
 */
public class Ejercicio5 {
    public static void main(String[] args) throws Exception {
        Scanner leer = new Scanner(System.in);
        
        try{
            System.out.print("Ingrese el valor 1: ");
            int valor1 = leer.nextInt();
            System.out.print("Ingrese el valor 2: ");
            int valor2 = leer.nextInt();
            
            if ((valor1 < 0) || (valor2 < 0)) {
                throw new Exception("ERROR: no se permite ingresar números negativos");
            }
            
            if (valor1 > 20 ) {
                throw new Exception("ERROR: el primer valor no debe ser mayor a 20");
            }
            
            double result = valor1/valor2;
            
            System.out.println("El resultado es: "+result);
        }
        
        catch(Exception e){
            System.out.println("Ha ocurrido una excepción de tipo "+e);
        }
    }
}

File: Segundo-Bimestre/Parcial 2/Unidad9-Exceptions/Unidad9-Exceptions_python/Ejercicio5.py
print("")
# Inicializar las variables
valor1 = 0; valor2 = 0; resultado = 0

# Proceso
try:
    valor1 = int(input("Ingrese el valor 1: ")) 
    valor2 = int(input("Ingrese el valor 2: ")) 

    if ((valor1 < 0) or (valor2 < 0)):
        print("")
        raise print("ERROR: no se permite ingresar números negativos")
    
    if (valor1 > 20 ):
        print("")
        raise print("ERROR: el primer valor no debe ser mayor a 20")
    
    resultado = valor1/valor2

    print("El resultado es: ",resultado)

except Exception as e:
    print("")
    print("Ha ocurrido una excepción de tipo",e)
    print("")